### PR TITLE
fix task bug

### PR DESCRIPTION
**Pull request by MayElnathan**

### CODE REVIEW

File: src/main/service.js [hunk 1]
```diff
@@ -362,10 +362,13 @@ function getMetaQuestionForAppendix(req, res, next){
...
-        req.log.info("request to create metaQuestion");
-        res.send(200, {code:200, metaQuestion})
-        next()
+        application.addMetaQuestion(req.body).then(
+            metaQuestion => {
+                req.log.info("request to create metaQuestion");
+                res.send(200, {code:200, metaQuestion})
+                next()
+            }
+        )
```
Comment (RoiTief) [inline src/main/service.js:371]: you don't encapsulate an `async` call in `try/catch`
instead you pass it 2 callbacks: 
1. first for success, the argument it takes is the returned value (undefined for void funcs).
2. second for if an error occurs, the argument is the thrown error. 

take a look at other `then`s I implemented in this file

File: src/main/business/applicationFacade.js [hunk 2]
```diff
@@ -283,6 +283,7 @@ class ApplicationFacade{
      * @throws {Error} - if there is no user logged in data
      */
     async viewMyTasks(data){
+        validateParameters(data,{})
         const user = await this.userController.getUser(data.callingUser.username);
         return this.taskController.getTasksOf(user.getUsername());
```
Comment (RoiTief) [inline src/main/business/applicationFacade.js:288]: Wouldn't it be easier to just call the `taskController` with `data` and extract the username from `callingUser` as part of the logic of `getTaskOf`?

File: src/main/business/TaskManager/Task.js [hunk 3]
```diff
@@ -13,17 +13,17 @@ const TaskPriority = {
...
         this.properties = addTaskProperty.properties ?? {}
-        this.description = addTaskProperty.description ?? null
+        this.description = addTaskProperty.description 
         this.assignedUsers = addTaskProperty.assignedUsers ?? []
-        this.action = addTaskProperty.action ?? null
-        this.finished = addTaskProperty.false ?? null
-        this.options = addTaskProperty.options ?? null
-        this.response = addTaskProperty.response ?? null,
+        this.action = addTaskProperty.action 
+        this.finished = addTaskProperty.finished ?? false
```
Comment (RoiTief) [inline src/main/business/TaskManager/Task.js:24]: what is `??`

File: src/main/business/MetaQuestions/MetaQuestionController.js [hunk 4]
```diff
@@ -29,8 +29,8 @@ class MetaQuestionController{
         }
         let metaQuestion = new MetaQuestion(data);
         this.#metaQuestions.set(this.#metaQuestionId, metaQuestion);
-        const ta_s = this.#userController.getAllStaff(data)["TAs"]
-
+        let ta_s = await this.#userController.getAllStaff(data)
+        ta_s = ta_s["TAs"]
```
Comment (RoiTief) [inline src/main/business/MetaQuestions/MetaQuestionController.js:33]:
```suggestion
        let ta_s = (await this.#userController.getAllStaff(data))["TAs"];
```

File: src/main/business/MetaQuestions/MetaQuestionController.js [hunk 5]
```diff
@@ -40,10 +40,11 @@ class MetaQuestionController{
         return metaQuestion
     }
 
-    editMetaQuestion(data) {
+    async editMetaQuestion(data) {
         let metaQuestion = new MetaQuestion(data);
         this.#metaQuestions.set(data.id, metaQuestion);
-        const ta_s = this.#userController.getAllStaff(data)["TAs"]
+        let ta_s = await this.#userController.getAllStaff(data)
+        ta_s =ta_s["TAs"]
```
Comment (RoiTief) [inline src/main/business/MetaQuestions/MetaQuestionController.js:47]:
```suggestion
        let ta_s = (await this.#userController.getAllStaff(data))["TAs"];
```

File: ReactApp/src/utils/rest-api-call.js [hunk 6]
```diff
@@ -39,6 +40,11 @@ export const serverPath = {
     REFRESH_TOKEN: 'refreshJWT',
 }
 
+
+export const pathToReturnTypeMap={
+  [serverPath.VIEW_TASKS]: {tasks:[{finished:PRIMITIVE_TYPES.BOOLEAN, }]},
+}
```
Comment (RoiTief) [inline ReactApp/src/utils/rest-api-call.js:46]: What is that? Where did you use it?

File: src/main/business/TaskManager/Task.js (hunk 3 above)
Comment (OfekN11) [reply src/main/business/TaskManager/Task.js:24]: If it is null or undefined put the value

File: ReactApp/src/utils/rest-api-call.js (hunk 6 above)
Comment (OfekN11) [reply ReactApp/src/utils/rest-api-call.js:46]: My bad I forgot to change it, anyway it is more orginize way to follow on the api expected return types

File: src/main/business/applicationFacade.js (hunk 2 above)
Comment (OfekN11) [reply src/main/business/applicationFacade.js:288]: Nope there isnt...

File: ReactApp/src/utils/rest-api-call.js (hunk 6 above)
Comment (OfekN11) [reply ReactApp/src/utils/rest-api-call.js]: Please check it now

File: ReactApp/src/sections/overview/overview-my-tasks.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import PropTypes from 'prop-types';
-import { httpsMethod, serverPath, requestServer } from 'src/utils/rest-api-call';
+import { httpsMethod, serverPath, requestServer, pathToReturnTypeMap } from 'src/utils/rest-api-call';
```
Comment (RoiTief) [inline ReactApp/src/sections/overview/overview-my-tasks.js:2]: you import it but don't use it?

File: ReactApp/src/utils/rest-api-call.js [hunk 8]
```diff
@@ -121,9 +128,11 @@ export async function requestServer(path, method, body, expectedResponseType) {
                 body: JSON.stringify(body)
             })
     }
+
     const retObject = await extractDataFromResponse(response)
-    if(expectedResponseType){
-      validateParameters(retObject, expectedResponseType,false, false)
+    
+    if(pathToReturnTypeMap[path]){
+      validateParameters(retObject, pathToReturnTypeMap[path], false, false)
```
Comment (RoiTief) [inline ReactApp/src/utils/rest-api-call.js:135]: ok I get this now.

File: src/main/service.js [hunk 9]
```diff
@@ -192,7 +192,7 @@ function viewMyTasks(req, res, next){
  */
 function finishATask(req, res, next) {
     try{
-        application.finishATask(data).then(()=>{
+        application.finishATask(req.body).then(()=>{
```
Comment (RoiTief) [inline src/main/service.js:195]: fix this `try/catch` as well

File: src/main/service.js [hunk 10]
```diff
@@ -399,10 +399,13 @@ function addMetaQuestion(req, res, next){
  */
 function editMetaQuestion(req, res, next){
     try{
-        application.editMetaQuestion(req.body)
-        req.log.info("request to edit metaQuestion");
-        res.send(200, {code:200})
-        next()
+        application.editMetaQuestion(req.body).then(
```
Comment (RoiTief) [inline src/main/service.js:402]: fix this `try/catch` as well

File: ReactApp/src/sections/overview/overview-my-tasks.js (hunk 7 above)
Comment (OfekN11) [reply ReactApp/src/sections/overview/overview-my-tasks.js]: KM

**Review by RoiTief — APPROVED** [review]

**Review by RoiTief — APPROVED** [review]

**Pull request merged**